src/main/java/cubicchunks/lighting/LightingManager.java

```diff
@@ -214,16 +209,66 @@ void markBlockColumnForUpdate(int localX, int localZ) {
         }
 
         public void tick() {
+            LightUpdateTracker tracker = cube.getCubicWorld().getLightingManager().getTracker();
+            for(EnumFacing dir:EnumFacing.values()){
+                if(cube.edgeNeedSkyLightUpdate[dir.ordinal()]){
+                    ICubeProvider cache = cube.getCubicWorld().getCubeCache();
+                    CubePos cpos = cube.getCoords();
+                    Cube loadedCube = cache.getLoadedCube(
+                            cpos.getX() + dir.getFrontOffsetX(), 
+                            cpos.getY() + dir.getFrontOffsetY(), 
+                            cpos.getZ() + dir.getFrontOffsetZ());
```

Comment:
To avoid doing some of the work twice, it may be better to get the neighbor cube before checking the edgeNeedsUpdate flag, and `||` that flag with the flag of neighbor cube if it's loaded.

Reply:
I would like to keep `cache.getLoadedCube(...)` behind `if (cube.edgeNeedSkyLightUpdate[dir.ordinal()])` because getting cube from XYZ map is quite slow operation.

Reply:
Hm... good point. Then at least also reset the corresponding flag in the neighbor cube, and do a both-ways spread if it's also true (16x16x2 instead of 16x16x1 blocks).